### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by luii**
@piranna Im done with the documentation. If you want to review the documentation this is okay for me.
Or just merge it.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -39,6 +43,21 @@ function onerror(error)
...
 }
 
+/**
+ * This functions takes the cmdline **showed below in the example** and splits
+ * it into key value pairs
+ * @access  private
+ * @param   {String} cmdline This string contains information about the
+ *                           initrd and the root partition
+ * @return  {Object}         It returns a object containing key value pairs
+ *                           if no value is provided, then its just true
```
Comment (piranna) [inline server.js]: Not clear about what is "it, put "if no value is provided for a cmdline argument, then it's set to true"

File: server.js [hunk 2]
```diff
@@ -93,6 +160,16 @@ function mountUserFilesystems(arr, upperdir, callback)
...
 
+/**
+ * Waits until dev is mounted and then executes `mountUserFilesystems` to
+ * mount `$upperdir/proc` and `$upperdir/tmp`
+ * @access private
+ * @param  {String}   upperdir
+ * @param  {Boolean}  isRoot   True if the user is root, false if not
+ * @param  {Function} callback The callback function
+ * @return {Function}          The callback is either called with a error
+ *                             or if the user filesystems are mounted
```
Comment (piranna) [inline server.js]: return is to describe the return of a function, this description is for the callback (look for callback in JsDoc)

File: server.js [hunk 3]
```diff
@@ -207,11 +293,26 @@ function overlay_user(usersFolder, user, callback)
   })
 }
 
+/**
+ * Filters users
+ * If the first char is not a dot (invisible folder?) and not root and not
+ * ´lost+found´ return true
```
Comment (piranna) [inline server.js]: "Filter folders that are valid users $HOME" is simpler and easier to understand ;-)

File: server.js [hunk 4]
```diff
@@ -279,6 +411,12 @@ function askLocation(error)
   prompt.get('path to userfs', pathToUserfs)
 }
 
+/**
+ * If the single key is set in the cmdline it starts a admin repl
```
Comment (piranna) [inline server.js]: Put `single` and other code words between backticks (`) so they are shown as code.

**Comment by luii** [comment]
@piranna Updated the whole documentation according to your review!

**Comment by piranna** [comment]
Merged :-) Since it's only docs, I think I'll left the npm publish gor the next code future.

**Pull request merged**